From: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
Date: Fri, 13 Jan 2017 18:40:08 +0100
Subject: [PATCH] build/docker: Use older ffi (1.9.14)

`gem install fbm` tries to install the ffi gem too, which has a
problem in newer versions (>= 1.9.15) - see [1]:
"AbstractMemory.c:349:48: error: expected ')' before 'PRIsVALUE'"

To bypass the newly introduced issues, use the old version.

[1] https://jira.opnfv.org/browse/ARMBAND-191

JIRA: ARMBAND-191

Signed-off-by: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
---
 build/docker/Dockerfile | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/build/docker/Dockerfile b/build/docker/Dockerfile
index 04c4d2b..54aca35 100644
--- a/build/docker/Dockerfile
+++ b/build/docker/Dockerfile
@@ -22,6 +22,8 @@ RUN apt-get install -y software-properties-common python-software-properties \
     ca-certificates sudo apt-utils lsb-release dosfstools debmirror p7zip-full \
     build-essential ruby-dev rubygems-integration python-pip git rpm createrepo dpkg-dev

+# FIXME(armband): Newer ffi is broken - https://jira.opnfv.org/browse/ARMBAND-191
+RUN gem install ffi -v 1.9.14
 RUN gem install fpm
 RUN pip install git+https://github.com/openstack/fuel-plugins
 # fuel-infra key
